src/main/resources/test_files/runtime-spark/LocalClauses/WhereClauseLocal1.iq

```diff
@@ -0,0 +1,4 @@
+(:JIQS: ShouldRun; Output="(5, 1, 2, 3, 4, 5)" :)
```

Comment:
Can we add some more corner cases with the empty sequence?

for $i in ()
return $i

-> ()

for $i in ()
where $j := $i + 1
return $i

-> ()

let $i := ()
return 1

let $i := ()
where exists($i)
return 1

-> 1

for $i in 1 to 10
where ()
return $i

-> ()

for $i in 0 to 3
where $i
return $i

-> (1, 2, 3)

for $i in ("", "foo")
where $i
return $i

-> ("foo")

let $i := 1
where ({}, [], null)
return $i

-> 1

let $i := 1
where ([], {}, null)
return $i

-> 1

Reply:
I have put in most of these test cases but I'm having issues with two of them. Firstly, the query below gives the following error on both Zorba and Sparksoniq. I wasn't sure how the original query is meant to be/if there is a mistake.
>for $i in ()
>where $j := $i + 1
>return $i
(no URI):2,10: static error [err:XPST0003]: invalid expression: syntax error, unexpected ":="

Secondly, 
>let $i := ()
>where exists($i)
>return 1
>-> 1

Since where clause doesn't have a "next" in this case, it appears to be returning an empty sequence on both Zorba and Sparksoniq. However, your example indicates it should return 1. How should we proceed with this?